cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1444,6 +1467,41 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
             return Future.succeededFuture(this);
         }
 
+        Future<ReconciliationState> kafkaPersistentClaimDeletion() {
+            if (kafkaCluster.getStorage() instanceof PersistentClaimStorage) {
```

Comment:
I see the block of code here, pretty much duplicating the code for the ZK case, and think we should factor out a common method for the two. I know ZK doesn't support JBOD, but we don't really care at this point in the code: Both call ZK and Kafka cases could become callsites to a common method which handled deletion for all storage types.

Reply:
Done